android/src/main/java/com/baseflow/flutter/plugin/geolocator/utils/MainThreadDispatcher.java

```diff
@@ -0,0 +1,45 @@
+package com.baseflow.flutter.plugin.geolocator.utils;
+
+import android.os.Handler;
+import android.os.Looper;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.baseflow.flutter.plugin.geolocator.data.wrapper.ChannelResponse;
+
+public class MainThreadDispatcher {
+    private static Handler handler = new Handler(Looper.getMainLooper());
+
+    public static void dispatch(Runnable runnable) {
+        initHandlerIfNull();
+        handler.post(runnable);
+    }
+
+    private static void initHandlerIfNull() {
+        if (handler == null) {
+            handler = new Handler(Looper.getMainLooper());
+        }
+    }
+
+    public static void dispatchError(@NonNull final ChannelResponse channelResponse, @NonNull final String channelName,
+                                     @NonNull final String error, @Nullable final Object details) {
+        initHandlerIfNull();
+        handler.post(new Runnable() {
+            @Override
+            public void run() {
+                channelResponse.error(channelName, error, details);
+            }
+        });
+    }
+
+    public static void dispatchGeocodeResult(@NonNull final ChannelResponse channelResponse, final Object result) {
```

Comment:
I think a better name here would be `dispatchSuccess` as it could theoretically communicate any kind of success message (not only geocoding results). This would potentially make is easier to reuse the dispatcher.

```suggestion
    public static void dispatchSuccess(@NonNull final ChannelResponse channelResponse, final Object result) {
```

Reply:
Yeah you are right, its much better, i let the `dispatch` method for general use, but this naming is much better. We can leave `dispatch` method for even more generic use.